File: src/main/java/com/sidgate/deck/Rank.java
package com.sidgate.deck;

import static java.util.Arrays.stream;

public enum Rank {
	TWO("2"),
	THREE("3"),
	FOUR("4"),
	FIVE("5"),
	SIX("6"),
	SEVEN("7"),
	EIGHT("8"),
	NINE("9"),
	TEN("T"),
	JACK("J"),
	QUEEN("Q"),
	KING("K"),
	ACE("A");

	private String name;

	Rank(String name) {
		this.name = name;
	}

	public static Rank getByShortName(String name) {
		return stream(values()).filter(rank -> rank.name.equals(name)).findFirst().get();
	}

	public String getShortName() {
		return name;
	}
}

File: src/main/kotlin/sidgate/deck/Rank.kt
package sidgate.deck

import java.util.*

enum class Rank(val shortName: String) {
    TWO("2"), THREE("3"), FOUR("4"), FIVE("5"), SIX("6"), SEVEN("7"), EIGHT("8"), NINE("9"), TEN("T"), JACK("J"), QUEEN("Q"), KING("K"), ACE("A");

    companion object {
        fun getByShortName(name: String): Rank {
            return Arrays.stream(values()).filter { rank: Rank -> rank.shortName == name }.findFirst().get()
        }
    }

}